### PR TITLE
savegame: fixed Lara's outfit in Lara's Home when replaying the level

### LINKED ISSUE #571: T1M bug: glitched gym outfit
Starting the gym, completing it and starting it again causes Lara to wear her regular adventure outfit.
I believe this has to do with 2.9 and Start Info refactors.

@walkawayy could you please take a look?

### PR DESCRIPTION
**Pull request by walkawayy**
Resolves #571.

#### Checklist

- [X] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description
Fixed Lara's outfit in Lara's Home when replaying the level.
...

### CODE REVIEW

**Comment by walkawayy** [comment]
Random aside I noticed: I don't think `RESUME_INFO resume->flags.available` is ever used.

**Comment by rr-** [comment]
I think I'd prefer if we called `Savegame_InitCurrentInfo` in `Game_Start` in case `level_type == GFL_GYM`.

```diff
diff --git a/src/game/game/game.c b/src/game/game/game.c
index 1a85d6eb..1117583c 100644
--- a/src/game/game/game.c
+++ b/src/game/game/game.c
@@ -194,6 +194,14 @@ bool Game_Start(int32_t level_num, GAMEFLOW_LEVEL_TYPE level_type)
         }
         break;
 
+    case GFL_GYM:
+        Savegame_InitCurrentInfo();
+        Level_InitialiseFlags();
+        if (!Level_Initialise(level_num)) {
+            return false;
+        }
+        break;
+
     default:
         Level_InitialiseFlags();
         if (!Level_Initialise(level_num)) {
```

(untested)

(As an aside it'd be nice to have `GFL_NEWGAME` `GFL_NEXTLEVEL` or similar but less dumb-sounding consts, to tell apart the case when the player starts the game for the first time from the case where the player continues the game onto the next level)

**Comment by walkawayy** [comment]
Ok I can try to add that. I wasn't sure what you wanted to do. I tried to match `Savegame_ApplyLogicToCurrentInfo` which has checks for Gym and first level.

File: src/global/types.h [hunk 1]
```diff
@@ -993,6 +993,7 @@ typedef enum GAMEFLOW_OPTION {
     GF_START_SAVED_GAME = 7 << 6,
     GF_RESTART_GAME = 8 << 6,
     GF_SELECT_GAME = 9 << 6,
+    GF_START_GYM = 10 << 6,
```
Comment (rr-) [inline src/global/types.h:996]: That's not what I meant. This enum is not necessary; I meant to discern between **new game** and **continue to next level** on the `GFL` level. (Both of which are out of scope for this pull request)

Comment (rr-) [reply src/global/types.h:996]: On another thought let's keep things this way, they're pretty clean.

**Review by rr- — APPROVED** [review]

**Pull request merged**